osquery/config/plugins/http_tests.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+#include <thread>
+#include <iostream>
+#include <sstream>
+
+#include <boost/network/protocol/http/server.hpp>
+#include <boost/property_tree/ptree.hpp>
+
+#include <gtest/gtest.h>
+
+#include <osquery/config.h>
+#include <osquery/enrollment.h>
+
+#include "osquery/remote/requests.h"
+#include "osquery/remote/transports/http.h"
+#include "osquery/remote/serializers/json.h"
+
+namespace http = boost::network::http;
+
+namespace osquery {
+
+DECLARE_string(enrollment_uri);
+DECLARE_string(configuration_uri);
+DECLARE_string(app_id);
+
+class HttpConfigTests : public testing::Test {
+ public:
+  void SetUp() {}
+};
+
+struct EnrollHTTPHandler;
+struct ConfigHTTPHandler;
+typedef http::server<EnrollHTTPHandler> EnrollServer;
+typedef http::server<ConfigHTTPHandler> ConfigServer;
+
+struct EnrollHTTPHandler {
+  void operator()(EnrollServer::request const &request,
+                  EnrollServer::response &response) {
+    response = EnrollServer::response::stock_reply(
+        EnrollServer::response::ok,
+        std::string("{\"enrollment_key\":\"potatoes\"}"));
+  }
+  void log(...) {}
+};
+
+struct ConfigHTTPHandler {
+  void operator()(ConfigServer::request const &request,
+                  ConfigServer::response &response) {
+    response = ConfigServer::response::stock_reply(
+        ConfigServer::response::ok,
+        std::string(
+            "{ \"schedule\": { \"launchd\": { \"query\": \"select * from "
+            "launchd;\", \"interval\": 3600 }, \"all_kexts\": { \"query\": "
+            "\"select name, version from kextstat;\", \"interval\": 600 } } "
+            "}"));
+  }
+  void log(...) {}
```

Comment:
Yeah, required by contract, otherwise
```
osquery/git/osquery/third-party/cpp-netlib/boost/network/protocol/http/server/sync_connection.hpp:57:16: error:
      no member named 'log' in 'osquery::ConfigHTTPHandler'```